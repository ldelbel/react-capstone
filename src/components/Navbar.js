import React from 'react';

const Navbar = () => (
  <nav className="navbar navbar-light navbar-expand-md" style={{ background: 'var(--cyan)' }}>
    <div className="container-fluid">
      <a className="navbar-brand" href="/" style={{ fontFamily: 'Alfa Slab One, cursive', color: 'var(--white)', fontSize: '35px' }}>Stonk$</a>
      <button type="button" data-toggle="collapse" className="navbar-toggler" data-target="#navcol-1">
        <span className="sr-only">Toggle navigation</span>
        <span className="navbar-toggler-icon" />
      </button>
      <div className="collapse navbar-collapse" id="navcol-1">
        <ul className="nav navbar-nav">
          <li className="nav-item"><a className="nav-link active" href="/">First Item</a></li>
          <li className="nav-item"><a className="nav-link" href="/">Second Item</a></li>
          <li className="nav-item"><a className="nav-link" href="/">Third Item</a></li>
        </ul>
      </div>
    </div>
  </nav>
);

export default Navbar;
