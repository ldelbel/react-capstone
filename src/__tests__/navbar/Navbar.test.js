import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Navbar from '../../pages/common/Navbar';

describe('Navbar DOM', () => {
  it('renders correctly', () => {
    const navbar = renderer.create(
      <Navbar />,
    ).toJSON();
    expect(navbar).toMatchSnapshot();
  });

  it('has brand name as first element', () => {
    const { container } = render(<Navbar />);
    expect(container.firstChild.firstChild).toHaveTextContent('StockList');
  });
});
