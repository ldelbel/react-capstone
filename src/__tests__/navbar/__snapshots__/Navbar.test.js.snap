// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar DOM renders correctly 1`] = `
<nav
  className="navbar navbar-light navbar-expand-md"
  style={
    Object {
      "background": "var(--cyan)",
    }
  }
>
  <div
    className="container-fluid"
  >
    <a
      className="navbar-brand"
      href="/"
      style={
        Object {
          "color": "var(--white)",
          "fontFamily": "Alfa Slab One, cursive",
          "fontSize": "35px",
        }
      }
    >
      StockList
    </a>
    <button
      className="navbar-toggler"
      data-target="#navcol-1"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="sr-only"
      >
        Toggle navigation
      </span>
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navcol-1"
    >
      <ul
        className="nav navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link active"
            href="/"
            style={
              Object {
                "color": "white",
                "marginLeft": "50px",
              }
            }
          >
            Home
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
